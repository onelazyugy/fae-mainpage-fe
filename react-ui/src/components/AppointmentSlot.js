import React from "react";
import Paper from 'material-ui/Paper';

const AppointmentSlot = ({slots, onAppointmentSlotClicked}) => {
    const styleNormal = {
        height: 100,
        width: 100,
        margin: 20,
        textAlign: 'center',
        display: 'inline-block',
        cursor: "pointer"
    };
    const styleBooked = {
        height: 100,
        width: 100,
        margin: 20,
        textAlign: 'center',
        display: 'inline-block',
        cursor: "pointer",
        backgroundColor: "red"
    };
    let apptTimeSlots = [];
    slots.map((slot, index)=>{
        let style;
        if(slot.isSlotTaken) {
            style = styleBooked;
        } else {
            style = styleNormal;
        }
        apptTimeSlots.push(<Paper key={index} id={index} style={style} zDepth={1} onClick={onAppointmentSlotClicked}>{slot.time}</Paper>);
    });
    return (
        <div>{apptTimeSlots}</div>
    );
};

export default AppointmentSlot;