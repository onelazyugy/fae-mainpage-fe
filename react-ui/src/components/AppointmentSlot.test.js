import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import AppointmentSlot from './AppointmentSlot';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'

const props = {
    slots: [{
        "id": 0,
        "time": "9 am",
        "isSlotTaken": false,
        "name": "",
        "phone": ""
    },
    {
        "id": 1,
        "time": "10 am",
        "isSlotTaken": false,
        "name": "",
        "phone": ""
    }]
};

describe('AppointmentSlot', () => {
    it('renders a snapshot', () => {
        const tree = renderer.create(<MuiThemeProvider><AppointmentSlot {...props} /></MuiThemeProvider>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    let appointmentSlot = mount(<MuiThemeProvider><AppointmentSlot {...props} /></MuiThemeProvider>);
    it('has 2 slots', () => {
        // console.log('appointmentSlot:', appointmentSlot.debug());
        // console.log('div 1 under paper', appointmentSlot.find('Paper'));
        // console.log('div 1 under paper text', appointmentSlot.find('div').at(1).text());
        expect(appointmentSlot.find('Paper').length).toEqual(2);
    });

    it('render 9 am slot', () => {
        // console.log('appointmentSlot:', appointmentSlot.debug());
        // console.log('div 1 under paper', appointmentSlot.find('div').at(1).debug());
        // console.log('div 1 under paper text', appointmentSlot.find('div').at(1).text());
        expect(appointmentSlot.find('div').at(1).text()).toEqual('9 am');
    });

    it('render 10 am slot', () => {
        // console.log('appointmentSlot:', appointmentSlot.debug());
        // console.log('div 1 under paper', appointmentSlot.find('div').at(1).debug());
        // console.log('div 1 under paper text', appointmentSlot.find('div').at(1).text());
        expect(appointmentSlot.find('div').at(2).text()).toEqual('10 am');
    });
});