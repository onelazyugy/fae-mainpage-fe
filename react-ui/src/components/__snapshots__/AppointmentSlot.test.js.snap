// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentSlot renders a snapshot 1`] = `
<div>
  <div
    id={0}
    onClick={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": 2,
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "height": 100,
        "margin": 20,
        "muiPrepared": true,
        "textAlign": "center",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 100,
      }
    }
  >
    9 am
  </div>
  <div
    id={1}
    onClick={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": 2,
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "height": 100,
        "margin": 20,
        "muiPrepared": true,
        "textAlign": "center",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 100,
      }
    }
  >
    10 am
  </div>
</div>
`;
